text/0055-sort.md

```diff
@@ -0,0 +1,424 @@
+- Title: Sort
+- Start Date: 2021-07-20
+- Specification PR: [#55](https://github.com/meilisearch/specifications/pull/55)
+- Discovery Issue: [#43](https://github.com/meilisearch/product/issues/43)
+- MeiliSearch Tracking-issues:
+
+# Sort
+
+## 1. Functional Specification
+
+### I. Summary
+
+The purpose of this specification is to add a **sort** feature at search time to quickly sort the search results as an end-user. Fields called `sortable-attributes` must be known for the search to be usable. These fields can be of type `string` and `number`. We have also introduced a new ranking rule called `sort`, allowing the user to adjust how the sorting should behave. Its position within the ranking rules allows to adjust its behavior according to the needs of exhaustivity and relevancy. We also introduced a `sort` parameter on the search resource to give the end-user the ability to sort the search results according to his needs.
+
+#### Summary Key points
+
+- `sortable-attributes` setting MUST be known by the engine before search time.
+- `sort` search parameter MUST be able to operate on multiple fields at search time.  e.g. sort="price:asc,label:desc,...". `string` and `number` fields MUST be supported.
```

Comment:
I used the term "and" to specify that both types must be supported by the `sort` feature.

As said in the [first working group](https://github.com/meilisearch/product/issues/43#issuecomment-875715587) where we identified the do and don't for this first iteration, sort on nested field will not be supported. I mentioned that specific case in the Future Possibilities part at the end of the specification.

Similarly to the behavior we have on `filter` when at search time if a field is not known to be a filterableAttribute an `invalid_filter` error is returned to the user. It will be the same with the `sort` parameter, except that in this case, the user will receive an `invalid_sort` error.